http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelperSpec.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.workbench.leonardo.util
+
+import cats.effect.IO
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import com.google.cloud.compute.v1.Firewall
+import org.broadinstitute.dsde.workbench.google.GoogleProjectDAO
+import org.broadinstitute.dsde.workbench.google.mock.MockGoogleProjectDAO
+import org.broadinstitute.dsde.workbench.google2.FirewallRuleName
+import org.broadinstitute.dsde.workbench.leonardo.VPCConfig.{VPCNetwork, VPCSubnet}
+import org.broadinstitute.dsde.workbench.leonardo.dao.google.MockGoogleComputeService
+import org.broadinstitute.dsde.workbench.leonardo.{CommonTestData, LeonardoTestSuite}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.scalatest.FlatSpecLike
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+class VPCHelperSpec extends FlatSpecLike with LeonardoTestSuite {
+
+  "VPCHelper" should "get a subnet from a project label" in {
+    val test = new VPCHelper(CommonTestData.vpcHelperConfig,
+                             stubProjectDAO(
+                               Map(CommonTestData.vpcHelperConfig.projectVPCSubnetLabelName -> "my_subnet",
+                                   CommonTestData.vpcHelperConfig.projectVPCNetworkLabelName -> "my_network")
+                             ),
+                             MockGoogleComputeService)
```

Comment:
trying to understand the test, where is project label being passed? `CommonTestData.vpcHelperConfig` looks like is from leonardo's conf?

Reply:
The label keys are in leo conf -- we look up project labels given the keys. In prod Rawls sets the labels when it creates the project; here I've just stubbed the `GoogleProjectDAO`.